struct SimpleStruct
{
    1:    required string                  commonField1,
    2:    optional list<string>            commonField2,
    3:    required binary                  commonField3,
    4:    required bool                    commonField4,
    5:    required RestReference           commonField5,
    6:    optional double                  commonField6,
    7:    optional i32                     commonField7,
    8:    optional byte                    common_field8,
}
